sfge/src/main/java/com/salesforce/graph/symbols/apex/schema/DescribeSObjectResultFactory.java

```diff
@@ -3,136 +3,170 @@
-                        // EndLine='3' Image='Account' RealLoc='true'>
-                        //    <ReferenceExpression BeginColumn='19' BeginLine='3' Context=''
-                        // DefiningType='MyClass' EndLine='3' Image='Schema' Names='[Schema,
-                        // SObjectType]' RealLoc='true' ReferenceType='LOAD' SafeNav='false' />
-                        // </VariableExpression>
-                        // TODO: Is SObjectType.Account valid?
-                        int sObjectTypeIndex = chainedNames.size() - 1;
-                        int schemaIndex = chainedNames.size() - 2;
+                final List<String> chainedNames = vertex.getChainedNames();
+                final String vertexName = vertex.getName();
 
-                        String nameProperty = vertex.getName();
-                        String sObjectType = chainedNames.get(sObjectTypeIndex);
-                        String schema = chainedNames.get(schemaIndex);
+                final Optional<String> sObjectName =
+                        SObjectTypeChainUtil.getSObjectName(chainedNames, vertex);
 
-                        if (sObjectType.equalsIgnoreCase(
-                                ApexStandardLibraryUtil.VariableNames.S_OBJECT_TYPE)) {
-                            if (schema.equalsIgnoreCase(
-                                    ApexStandardLibraryUtil.VariableNames.SCHEMA)) {
-                                SObjectType apexValueSObjectType =
-                                        ApexValueBuilder.getWithoutSymbolProvider()
-                                                .buildSObjectType(nameProperty);
-                                return Optional.of(
-                                        ApexValueBuilder.getWithoutSymbolProvider()
-                                                .returnedFrom(apexValueSObjectType, null)
-                                                .buildDescribeSObjectResult(apexValueSObjectType));
-                            }
-                        }
+                if (sObjectName.isPresent()) {
+                    // SObjectType.MyObject__c.Name
+                    // Schema.SObjectType.MyObject__c.Name
+                    if (NAME_PROPERTY.equalsIgnoreCase(vertexName)) {
+                        // TODO: how do we handle other fields with individual types?
+                        //  Stretch goal when we get to parsing field definitions of SObjects
+                        return getApexStringValue(sObjectName.get());
+                    } else if (ApexStandardLibraryUtil.VariableNames.S_OBJECT_TYPE.equalsIgnoreCase(
+                            vertexName)) {
+                        // Schema.SObjectType.Account.SObjectType is a SObjectType value.
+                        //  Reject this as a {@link DescribeSObjectResult}
+                        // TODO: streamline all of this
+                        return Optional.empty();
                     }
```

Comment:
These are the kind of weird outliers that I'm wary of - glad we had a test to detect this. Added a skip case.